index.js

```diff
@@ -139,6 +139,7 @@ function session(options){
   store.generate = function(req){
     req.sessionID = generateId(req);
     req.session = new Session(req);
+    cookie.secure = cookie.secure || issecure(req, trustProxy);
```

Comment:
This will never work when serving requests on parallel both secure and not, because you are altering global state that belongs to multiple requests at the same time. This will have to be fixed before we can accept the PR.